### DISCUSSION TITLE
[V3] check component exists

### DISCUSSION THREAD

**Discussion by merzi** [livewire/livewire#7070] [1. Help]
In Version 2 of Livewire i checked with the getManifest Method of the LivewireComponentsFinder Class if a component exists.
But with Version 3 the LivewireComponentFinder Class has been removed.

Is there any possibility to get a list of all components or is there any method to check if a given component exists?

**Comment by PhiloNL** [comment]
There is no longer a way to get all Livewire components. If you know the directory where Livewire components are located you could use the filesystem to figure this out like this has also been done in the following upgrade command src/Features/SupportConsoleCommands/Commands/Upgrade/ChangeDefaultNamespace.php 

You can use the component registry to verify if a component exists src/Mechanisms/ComponentRegistry.php

**Comment by shahonseven** [comment]
I'm not sure if this is proper way or not but this is how I do it, it's a direct copy from https://github.com/livewire/livewire/blob/8e3d93cf22b86bf0f5b0befd2776cb9f33711377/src/Mechanisms/ComponentRegistry.php#L165C24-L165C45:

```
Blade::if('livewireExists', function (string $name) {
  $rootNamespace = config('livewire.class_namespace');

  $class = collect(str($name)->explode('.'))
    ->map(fn ($segment) => (string) str($segment)->studly())
    ->join('\\');

    if (! empty($rootNamespace)) {
      $class = '\\'.$rootNamespace.'\\'.$class;
    }

    return class_exists($class);
});
```
